### PR TITLE
 Add breakpoint and hit counters

### PR DESCRIPTION
**Pull request by tgunr**
Found these useful on occasion.

### CODE REVIEW

File: basis/tools/continuations/continuations.factor [hunk 1]
```diff
@@ -26,6 +26,47 @@ SYMBOL: break-hook
 
 \ break t "break?" set-word-prop
 
+SYMBOL: hit-counter
+
+: (hit-count-inc) ( -- )
+    hit-counter get dup
```
Comment (mrjbq7) [inline basis/tools/continuations/continuations.factor:32]: Probably should just be ``hit-counter inc``

File: basis/tools/continuations/continuations.factor [hunk 2]
```diff
@@ -26,6 +26,47 @@ SYMBOL: break-hook
...
+    [  1 + ]
+    [ drop 1 ] if
+    hit-counter set ;
+
+: hit-count-reset ( -- )
+    0 hit-counter set ;
```
Comment (mrjbq7) [inline basis/tools/continuations/continuations.factor:38]: I'm not sure about how this scoping will work, this sets it in the current namespace, and then the other words "get/set" it into the new namespace, so if your word does things like ``with-variable`` that binds a new namespace then maybe this won't work like you expect?

File: basis/tools/continuations/continuations.factor (hunk 1 above)
Comment (tgunr) [reply basis/tools/continuations/continuations.factor:32]: Yes, better name. I wrote this a long time ago.

Comment (mrjbq7) [reply basis/tools/continuations/continuations.factor:32]: Not just a name, you can use ``inc`` from namespaces

File: basis/tools/continuations/continuations.factor (hunk 2 above)
Comment (tgunr) [reply basis/tools/continuations/continuations.factor:38]: Yes, you're right, I just checked it doesn't work unless you remain in the current namespace. Suggestions?

Comment (tgunr) [reply basis/tools/continuations/continuations.factor:38]: On second thought, it's probably not worth pursuing. Just close the request. If I need it again will address it then.

**Pull request closed without merging**